daemon/volumes.go

```diff
@@ -226,12 +226,15 @@ func (daemon *Daemon) registerMountPoints(container *Container, hostConfig *runc
 			bind.Volume = v
 			bind.Source = v.Path()
 			// Since this is just a named volume and not a typical bind, set to shared mode `z`
-			if bind.Relabel == "" {
-				bind.Relabel = "z"
+			if bind.Mode == "" {
+				bind.Mode = "z"
 			}
 		}
 
-		if err := label.Relabel(bind.Source, container.MountLabel, bind.Relabel); err != nil {
+		if err := label.Relabel(bind.Source, container.MountLabel, bind.Mode); err != nil {
+			return err
+		}
+		if err := volume.FixUidGid(container.Config.User, bind.Source, bind.Mode); err != nil {
```

Comment:
So, if I understand right this is going on in host mount namespace. And if there is no such user on host(which is likely, it can also have another uid/gid) chown will be done to current user of docker daemon(most likely root). TLDR: I don't understand what this patch do.